### PR TITLE
Improve test coverage (WIP)

### PR DESCRIPTION
**Pull request by BobDotCom**
## Summary

Improves test coverage

## Checklist

- [ ] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.8" ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
```
Comment (ArjunSharda) [inline .github/workflows/pylint.yml:14]:
```suggestion
        uses: actions/setup-python@v4
```

File: .github/workflows/pylint.yml [hunk 2]
```diff
@@ -1,27 +1,27 @@
...
+on: [ push, pull_request ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ "3.8" ]
+    steps:
+      - uses: actions/checkout@v2
```
Comment (ArjunSharda) [inline .github/workflows/pylint.yml:12]:
```suggestion        
         - uses: actions/checkout@v3
```

**Review by ArjunSharda — COMMENTED** [review]
I've left some suggestions for bumping some versions.

File: .github/workflows/pylint.yml [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-#        run: |
-#          pylint $(git ls-files '*.py')
+name: Pylint
+
+on: [ push, pull_request ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
```
Comment (EmreTech) [inline .github/workflows/pylint.yml:8]: Why is a strategy matrix used here? This job is only being ran on one Python version

**Pull request closed without merging**